Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -217,6 +228,35 @@ def _getInputWorkspaces(self, runs, isTrans):
             workspaces.append(ws)
         return workspaces
 
+    def _sumBanks(self, workspaces, roiDetectorIDs):
+        for workspace_name in workspaces:
+            workspace = AnalysisDataService.retrieve(workspace_name)
+            is_group = isinstance(workspace, WorkspaceGroup)
+            if is_group:
+                workspace = workspace[0]
+
+            rect_detectors = workspace.getInstrument().findRectDetectors()
+            num_rect_detectors = len(rect_detectors)
+
+            if num_rect_detectors == 0:
+                continue
+
+            if num_rect_detectors != 1:
+                raise NotImplementedError(f"Not implemented for more than one rectangular detector, "
+                                          f"{num_rect_detectors} were found.")
+
+            # We don't sum banks for a linear detector
+            if rect_detectors[0].xpixels() == 1 or rect_detectors[0].ypixels() == 1:
+                continue
+
+            if is_group:
+                raise NotImplementedError("Not implemented for a WorkspaceGroup containing 2D detectors.")
+
+            args = {"InputWorkspace": workspace, "ROIDetectorIDs": roiDetectorIDs, "OutputWorkspace": workspace_name}
```

Comment:
There's a bug setting the output workspace because a child algorithm does not output its result to the ADS by default. (We could make it do that, but we don't really want to overwrite the input workspace). This means the workspace does not get overwritten. Subsequent functions rely on the output being in the ADS but because we're just returning the input workspace names the original input gets passed along, so the summed output here is ignored. 

Here's a script to demonstrate the reduction with and without the summing:

```
ReflectometryISISPreprocess(InputRunList='INTER45455_inst', OutputWorkspace='TOF_45455')

# With sum-banks step
ReflectometryISISSumBanks(InputWorkspace='TOF_45455', OutputWorkspace='TOF_45455_summed')
ReflectometryReductionOneAuto(InputWorkspace='TOF_45455_summed', ProcessingInstructions='22-27', ThetaIn=0.5, MomentumTransferMin=0.0064527921283059177, MomentumTransferStep=0.02017069966273068, MomentumTransferMax=0.072643025200858427, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_45455_summed', OutputWorkspace='IvsQ_45455_summed')

# Without sum-banks step
ReflectometryReductionOneAuto(InputWorkspace='TOF_45455', ProcessingInstructions='22-27', ThetaIn=0.5, MomentumTransferMin=0.0064527921283059177, MomentumTransferStep=0.02017069966273068, MomentumTransferMax=0.072643025200858427, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_45455_unsummed', OutputWorkspace='IvsQ_45455_unsummed', OutputWorkspaceWavelength='__TMP0x55838730ba00')

# Via workflow algorithm
ReflectometryISISLoadAndProcess(InputRunList='INTER45455_inst', ProcessingInstructions='22-27', ThetaIn=0.5, OutputWorkspaceBinned='IvsQ_binned_45455_workflow', OutputWorkspace='IvsQ_45455_workflow', OutputWorkspaceWavelength='__TMP0x55838730ba00')
match_summed, _ = CompareWorkspaces('IvsQ_binned_45455_workflow', 'IvsQ_binned_45455_summed')
match_unsummed, _ = CompareWorkspaces('IvsQ_binned_45455_workflow', 'IvsQ_binned_45455_unsummed')
print('Matches summed workspace = ', match_summed)
print('Matches unsummed workspace = ', match_unsummed)
```

You can see that the workflow algorithm matches the un-summed one:
```
Matches summed workspace =  False
Matches unsummed workspace =  True
```

Really we should not be relying on the ADS and we should just return the actual workspaces, but that's quite a big change which needs tidying up in general anyway. I think for now we can just add the summed workspace to the ADS with a new name, e.g. with a suffix `_summed_segments`, say. It would also be good to include this in the `TOF` group. We could do this in a similar way to the monitor workspaces [here](https://github.com/mantidproject/mantid/blob/main/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py#L107) - this just uses a known modified version of the input workspace name. Very hacky but all this needs to be fixed up properly!